src/login.rs

```diff
@@ -205,9 +218,17 @@ impl Login {
         }
     }
 
-    fn set_visible_page_name<W: WidgetExt>(&self, page_name: &str, widget_to_focus: &W) {
+    fn set_visible_page_name<W: WidgetExt>(
+        &self,
+        page_name: &str,
+        error_label: &gtk::Label,
+        widget_to_focus: &W,
+    ) {
         let self_ = imp::Login::from_instance(self);
 
+        // Before transition to the page, be sure to reset the error label because it still might
+        // conatain an error message from the time when it was previously visited.
+        error_label.set_label("");
```

Comment:
Not sure if this makes sense to be here. Wouldn't it better to reset the error label when sending the requests (so when pressing the next button)? This way the previous error goes away already after trying again.

Reply:
Yes. This will look very good when the user makes another incorrect input. This makes the error label more salient.
I tried to clean up the code a bit more, if you don't mind.

Reply:
I just tested this again and I think we need to do both.
(see my comment in the main thread)
With your suggestion, the user can either see how the error vanishes or another/a new error appears (high salience).
But given the situation:
1. user enters correct phone number
2. user enters wrong code -> error label is shown
3. user enters correct phone number again
4. user can see old code and old error message

Reply:
You're right.